src/context.rs

```diff
@@ -27,10 +28,14 @@ const MAX_SB_SQUARE: usize = (MAX_SB_SIZE * MAX_SB_SIZE);
 const INTRA_MODES: usize = 13;
 const UV_INTRA_MODES: usize = 13;
 
-const b_width_log2_lookup: [u8; 20] = [0, 0, 0,  0, 0, 1, 1, 1, 2, 2, 2, 3, 3, 3, 4, 4, 0, 2, 1, 3];
-const b_height_log2_lookup: [u8; 20] = [ 0, 0, 0, 0, 1, 0, 1, 2, 1, 2, 3, 2, 3, 4, 3, 4, 2, 0, 3, 1];
-const tx_size_wide_log2: [usize; 14] = [2, 3, 4, 5, 2, 3, 3, 4, 4, 5, 2, 4, 3, 5];
-const tx_size_high_log2: [usize; 14] = [2, 3, 4, 5, 3, 2, 4, 3, 5, 4, 4, 2, 5, 3];
+static mi_size_wide: [u8; BLOCK_SIZES_ALL] = [
+  1, 1, 2, 2, 2, 4, 4, 4, 8, 8, 8, 16, 16, 1, 4, 2, 8, 4, 16];
+static mi_size_high: [u8; BLOCK_SIZES_ALL] = [
+  1, 2, 1, 2, 4, 2, 4, 8, 4, 8, 16, 8, 16, 4, 1, 8, 2, 16, 4];
+const b_width_log2_lookup: [u8; BLOCK_SIZES_ALL] = [ 0, 0, 1, 1, 1, 2, 2, 2, 3, 3, 3, 4, 4, 0, 2, 1, 3, 2, 4];
+const b_height_log2_lookup: [u8; BLOCK_SIZES_ALL] = [ 0, 1, 0, 1, 2, 1, 2, 3, 2, 3, 4, 3, 4, 2, 0, 3, 1, 4, 2];
+const tx_size_wide_log2: [usize; TX_SIZES_ALL] = [2, 3, 4, 5, 2, 3, 3, 4, 4, 5, 2, 4, 3, 5];
+const tx_size_high_log2: [usize; TX_SIZES_ALL] = [2, 3, 4, 5, 3, 2, 4, 3, 5, 4, 4, 2, 5, 3];
```

Comment:
The original table in libaom is not usable as it is because of compile flags in C that are not compatible with Rust. In addition, I believe that rav1e eventually want its own tables in Rust, targeting minimal dependency on existing libaom.

Reply:
@ycho sure, but you do want to have `const FOO: &[usize] = &[0, 2, 3];` and not `const FOO: [usize] = [0, 1, 3]`. The former would copy the reference where `FOO` is used, the latter would put `[0, 1, 3]` verbatim.

Reply:
Thinking again, one of the reasons above is wrong. The C flag part. Because the compiled object file of C already reflected C flags, which rav1e has set accordingly.

Reply:
All right, after discussion and reading https://doc.rust-lang.org/book/first-edition/const-and-static.html, as lu-zero advised, realized I did wrong!
Adding '&' to const, or even using 'static' is correct way in rust.

Reply:
https://ibb.co/gu35xc
static and const &[..] = &[..] seems give same amount of asm code, while const only indeed generates more far more asm code,